ext/zpages/test/tracez_data_aggregator_benchmark.cc

```diff
@@ -0,0 +1,249 @@
+    std::thread err(StartEndSpansError, std::ref(tracer), num_spans / 3, true);
+    StartEndSpansLatency(tracer, num_spans / 3, true);
+    run.join();
+    err.join();
+  }
+}
+
+//////////////////// SAME BENCHMARKS, BUT USER VISITS WEBPAGE ////////////////////////
+
+BENCHMARK_DEFINE_F(TracezAggregator, BM_InstantSameGet)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    std::thread spans(StartEndSpans, std::ref(tracer), num_spans, false);
+    GetManyAggregations(std::ref(aggregator), num_spans);
+    spans.join();
+  }
+}
+
+BENCHMARK_DEFINE_F(TracezAggregator, BM_InstantUniqueGet)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    std::thread spans(StartEndSpans, std::ref(tracer), num_spans, true);
+    GetManyAggregations(std::ref(aggregator), num_spans);
+    spans.join();
+  }
+}
+
+BENCHMARK_DEFINE_F(TracezAggregator, BM_BucketsSameGet)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>> running_spans;
+    std::thread run(StartSpans, std::ref(running_spans), std::ref(tracer), num_spans / 3, false);
+    std::thread err(StartEndSpansError, std::ref(tracer), num_spans / 3, false);
+    std::thread lat(StartEndSpansLatency, std::ref(tracer), num_spans / 3, false);
+    GetManyAggregations(std::ref(aggregator), num_spans / 3);
+    run.join();
+    err.join();
+    lat.join();
+  }
+}
+
+BENCHMARK_DEFINE_F(TracezAggregator, BM_BucketsUniqueGet)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>> running_spans;
+    std::thread run(StartSpans, std::ref(running_spans), std::ref(tracer), num_spans / 3, true);
+    std::thread err(StartEndSpansError, std::ref(tracer), num_spans / 3, true);
+    std::thread lat(StartEndSpansLatency, std::ref(tracer), num_spans / 3, true);
+    GetManyAggregations(std::ref(aggregator), num_spans / 3);
+    run.join();
+    err.join();
+    lat.join();
+  }
+}
+
+//////////////////////////// RUN BENCHMARKS ///////////////////////////////
+
+BENCHMARK_REGISTER_F(TracezAggregator, BM_InstantSame)->Arg(10)->Arg(1000);
+BENCHMARK_REGISTER_F(TracezAggregator, BM_InstantSameGet)->Arg(10)->Arg(1000);
+BENCHMARK_REGISTER_F(TracezAggregator, BM_InstantUnique)->Arg(10)->Arg(1000);
+BENCHMARK_REGISTER_F(TracezAggregator, BM_InstantUniqueGet)->Arg(10)->Arg(1000);
+BENCHMARK_REGISTER_F(TracezAggregator, BM_BucketsSame)->Arg(10)->Arg(1000);
+BENCHMARK_REGISTER_F(TracezAggregator, BM_BucketsSameGet)->Arg(10)->Arg(1000);
+BENCHMARK_REGISTER_F(TracezAggregator, BM_BucketsUnique)->Arg(10)->Arg(1000);
+BENCHMARK_REGISTER_F(TracezAggregator, BM_BucketsUniqueGet)->Arg(10)->Arg(1000);
+
+BENCHMARK_MAIN();
```

Comment:
See this regarding benchmarking component that has side threads in it: https://github.com/google/benchmark#cpu-timers

Reply:
would the cpu timer be used to make sure the data aggregator, which is running in a period side thread, is measured? also, since all the helper functions are just making spans and not doing aggregation work, should i do pause and resume timing in those?

Reply:
I think it depends on the cpu-timer, which cpu timer do you think we should use in this case? 
Yes, all the setup functions shouldn't be accounted in the benchmark timing.